### ISSUE TITLE
VM console access is broken

### ISSUE THREAD

**Issue by DanielDent** [rancher/rancher#3721]
Rancher Version: 0.59.1
Docker Version: 1.10.2
OS: Debian Jessie

Steps to Reproduce: 
-> Enable KVM features in web UI
-> Launch a VM
-> Click 'open Console'

Results: 
-> Console modal pops up briefly and then disappears. Browser console has the following error message: "Msg: Must set host and portc.Error @ vendor-0a9885a0f5e0365fb5fc5964aa6b5af9.js:47
vendor-0a9885a0f5e0365fb5fc5964aa6b5af9.js:47"

Expected:
-> In previous versions, the console worked, even for VMs.

**Comment by liyimeng** [comment]
Hi Daniel, 
Is this a released feature?  I make similar try like you, but on RancherOS instead of Debian
I got this error 
<pre>
root@k-blade-3:/# docker logs 77763a74a51a

dnsmasq: bad dhcp-range at line 2 of /etc/dnsmasq.conf
/var/lib/rancher/startvm: line 99: let: NEWCIDR=16
16-1: syntax error in expression (error token is "16-1")
/var/lib/rancher/startvm: line 123: let: i=2886795267^(1<<16
16): missing `)' (error token is "16)")
Error: either "local" is duplicate, or "10.42.84.137/16" is a garbage.
Could not initialize SDL(No available video device) - exiting
</pre>

**Comment by liyimeng** [comment]
More update. If I run with 
Host OS: RancherOS, guest VM OS: rancher/vm-rancheros. I got the errror post above. 
Host OS: RancherOS, guest VM OS: rancher/vm-ubuntu. Works perfectly, do not have the issue your reported.

**Comment by deniseschannon** [comment]
This is not a released feature in Rancher yet. It's been exposed in the UI, but we haven't actually officially released support.

**Comment by DanielDent** [comment]
This functionality is way too useful and cool for me to wait for the official release for me to start using it :).

I understand it's not yet released/supported and thus breakage is a distinct possibility. However, if there is any information I can provide to help debug this and make the eventual release of this feature rock-solid, please let me know what I can do.

**Comment by vincent99** [comment]
@DanielDent Can you open up the (Chrome) browser dev tools -> Networking tab and capture the request and response for getting the console URL, and then the actual websocket connection opening (if there is one)?

**Comment by DanielDent** [comment]
@vincent99 Thanks for the hint. I noticed while I was there that the websocket connection stays in 'pending' state.

I have the rancher server behind an SSL reverse proxy (I've documented the config in an issue I opened a few months ago https://github.com/rancher/rancher.github.io/issues/152 - the config is pretty much the one recommended in your documentation).

I tried using SSH tunnels to bypass the nginx reverse proxy, and without the proxy in front of the websocket, the console works fine. 

I think it would be good to have the front end recognize failed websocket connections rather than having the console get hidden from the user.

When I have a few minutes more to debug the issue I'll investigate why it works without nginx but failed when going through nginx.
